project-base/storefront/pages/order/contact-information.tsx

```diff
@@ -87,6 +87,12 @@ const ContactInformationPage: FC<ServerSidePropsType> = () => {
         }
     }, []);
 
+    useEffect(() => {
+        if (currentCart.isCartEmpty || !currentCart.transport || !currentCart.payment) {
+            router.replace(currentCart.isCartEmpty ? cartUrl : transportAndPaymentUrl);
+        }
+    }, [currentCart.payment, currentCart.transport, currentCart.isCartEmpty]);
```

Comment:
I would vote for a better split of conditions. This seems a little bit weird to first check in the if and then conditionally change arguments. What do you think?

Reply:
Sure! Good idea.